helix-term/src/ui/spinner.rs

```diff
@@ -46,22 +55,25 @@ impl Spinner {
     }
 
     pub fn dots(interval: u64) -> Self {
-        Self::new(vec!["⣾", "⣽", "⣻", "⢿", "⡿", "⣟", "⣯", "⣷"], interval)
+        Self::new("⣾⣽⣻⢿⡿⣟⣯⣷", interval)
     }
 
     pub fn start(&mut self) {
         self.start = Some(Instant::now());
     }
 
-    pub fn frame(&self) -> Option<&str> {
-        let idx = (self
-            .start
-            .map(|time| Instant::now().duration_since(time))?
-            .as_millis()
-            / self.interval as u128) as usize
-            % self.count;
-
-        self.frames.get(idx).copied()
```

Comment:
If self.frames is Vec<String> this would be something like

```
self.frames.get(idx).map(|s| s.as_str())
```

And the return value should be &str